### ISSUE TITLE
MaskedArray ignores __array_priority__

### ISSUE THREAD

**Issue by abalkin** [numpy/numpy#5227]
Given a class like this:

```
class X(int):
    def __add__(self, other):
        return 42
    def __radd__(self, other):
        return 24
    __array_priority__ = 20
```

Adding to a numpy array works as expected:

```
>>> np.array(2) + X()
24
```

However, when I add `X()` to a masked array, `X.__radd__` is not called:

```
>>> np.ma.array(2) + X()
2
```

**Comment by charris** [comment]
The problem solved seems similar to #3907.

**Comment by mhvk** [comment]
This seems resolved in current master (with #5964 merged). @abalkin - can this be closed?

**Comment by jjhelmus** [comment]
This has been fixed, closing.  Thanks for the heads up and fix @mhvk.
